### ISSUE TITLE
Switch off the auto reset in dashboard settings?

### ISSUE THREAD

**Issue by houser42** [FlowFuse/node-red-dashboard#1340]
### Description

Is there a way to switch off the way the dashboard settings resets itself after every edit?
I can find no setting, is there one? If not this is a feature request.

### Have you provided an initial effort estimate for this issue?

I am no FlowFuse team member

**Comment by colinl** [comment]
I thought there was an open issue for this already, but apparently not.  It is indeed annoying that it keeps reverting to the default view.

**Comment by joepavitt** [comment]
Dashboard settings reset? What am I missing in terms of detail here?

**Comment by houser42** [comment]
> Dashboard settings reset? What am I missing in terms of detail here?

Make an edit in the 2.0 dashboard and have look what the GUI does after that edit?

**Comment by colinl** [comment]
Strange, I can't seem to replicate the issue I thought was there.  In particular, I thought that if one compressed some of the groups/pages, then made an edit and deployed, that the groups would all expand again, but that does not seem to be happening for me tonight.  There must be something more subtle going on.
@houser42 what exact symptom are you seeing?

**Comment by joepavitt** [comment]
Which part of the GUI are you referring to? The Dashboard itself? The Sidebar? The configuration options on a particular node?

**Comment by houser42** [comment]
I am referring to the behaviour of the sidebar open and close of pages and groups. It resets after an edit.

**Comment by colinl** [comment]
Are you sure you are using the latest version of the dashboard (1.17.1), and have you restarted node-red?  Which version of node-red are you using?
Can you give a detailed example of what you are doing, and show us screenshots of what happens.
What browser/version are you using?

**Comment by houser42** [comment]
Yes, I am using the latest version of dashboard  1.17.1.
I am restarting the computers quite often, as I am doing a lot of work on three similar installs.
The version of Node Red is 3.1.10. Can not change that as it comes with the OS and is updated when the OS updates.
It is Victrons Venus OS on one of their dedicated Linux Beagle bone boxes.
Will report again if the node red version changes on these boxes. Sorry for not providing this info at the outset, but I was not aware that I was reporting a possible bug at the time.

**Comment by colinl** [comment]
I don't think the node-red version is likely to be the problem, though @joepavitt might like to comment on that.
You still have not given us a detailed example of what you are seeing, or told us what browser you are using.  Can you try another browser.

**Comment by houser42** [comment]
I am sorry for sloppy communication. Travelling so, this is from an airport lounge.
It looks like all browsers on Linux are good and do not show this issue.
The only one on Mac that has the  issue is (predictably;) Safari. I have no Winbox here so can not comment on that.
I guess this one is for the back of the backlog cue ;) I'll use firefox.

**Comment by joepavitt** [comment]
I'm still not understanding what you mean by "Dashboard Settings"? Are you referring to the collapsable sections here in the sidebar:

<img width="615" alt="Screenshot 2024-09-30 at 09 46 53" src="https://github.com/user-attachments/assets/cc8121f4-92b3-4f1c-9e03-4b3369e5fac9">

**Comment by houser42** [comment]
YEs, that is what I was referring to. Apologies for late reply, hav been travelling.
It looks like a Safari only bug here. On Safari the sections will close after making any edit to a node.

**Comment by joepavitt** [comment]
> It looks like a Safari only bug here.

If I had a penny for every time someone said that to me 😁

**Comment by houser42** [comment]
Correction. 
This error can be similarly replicated on Firefox on Mac and Ubuntu also, although a little differently.

example o replicate:
select a textnode in a group in the V2.0 dashboard sidebar 
click focus to go to the node in a flow
make a copy of the node in place
watch the dashboard sidebar reset and close all pages and groups.

**Comment by colinl** [comment]
I can confirm that behaviour, except that the effect is to expand any collapsed pages/groups rather than collapse or close them.  Is that what you meant @houser42?
Also I notice that it seems impossible to Undo the text node duplication.

**Comment by joepavitt** [comment]
We understand the problem, we want to retain the expand/collapse state for each page/group - that is what I understand the request here to be.

**Comment by bartbutenaers** [comment]
@joepavitt, @Steve-Mcl 
Quick question.  To keep me going 😇

Do you guys perhaps have already any ideas in your head about *"to retain the expand/collapse state for each page/group in the sidebar"*?    If you could share your thoughts in short here, I might have a look at it (if not too complicated...).

Some questions that pop up in my head:
1. Needs this kind of info to be stored in the config node, which is linked to the sidebar?  Imho that is not worth all the complexity: i.e. it is enough to keep this info in memory, because in most cases you only need to persist that info within your current flow editor session.  That way it also won't result in a *Deploy"* button that becomes active, which might be confusing?

2. Where do I need to store that info?  If you have no idea yet, I will try to figure out on my own...

3. Other related info?

**Comment by Steve-Mcl** [comment]
I think this is simply an in-memory thing. Once an edit is made to a node and the call to refresh the tree is made, in theory, we should store the state of the tree, perform any necessary UI updates then restore the state of the tree. Or maybe there is a more intelligent way of updating the tree without rebuilding it every time. Something along those lines. (Written from phone, sorry for the lack of detail)

In short, from a user perspective, the tree should not be expanding or collapsing or changing after an edit. 

Good luck with this one Bart & and thank you very much for picking these annoying niggles up.

**Comment by houser42** [comment]
What @Steve-Mcl  writes makes sense to me so I guess this is just a concurring opinion.
This is of course a pretty important thing when you are doing a lot of work on the dashboard, so also grateful it is getting some attention. ... and best wishes for 2025.
